bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxAirPurifierHandler.java

```diff
@@ -0,0 +1,218 @@
+                        Units.MICROGRAM_PER_CUBICMETRE);
+            case CHANNEL_PM10:
+                return new QuantityType<>(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().getPm10(),
+                        Units.MICROGRAM_PER_CUBICMETRE);
+            case CHANNEL_CO2:
+                return new QuantityType<>(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().getCo2(),
+                        Units.PARTS_PER_MILLION);
+            case CHANNEL_FAN_SPEED:
+                return new StringType(Integer.toString(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().getFanspeed()));
+            case CHANNEL_FILTER_LIFE:
+                return new QuantityType<>(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().getFilterLife(),
+                        Units.PERCENT);
+            case CHANNEL_IONIZER:
+                return OnOffType.from(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().isIonizer());
+            case CHANNEL_UI_LIGHT:
+                return OnOffType.from(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().isUiLight());
+            case CHANNEL_SAFETY_LOCK:
+                return OnOffType.from(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().isSafetyLock());
+            case CHANNEL_WORK_MODE:
+                return new StringType(
+                        ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().getWorkmode());
+            case CHANNEL_DOOR_OPEN:
+                return ((AirPurifierStateDTO) dto.getApplianceState()).getProperties().getReported().isDoorOpen()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED;
+            case CONNECTION_STATE:
+                return "Connected".equals(((AirPurifierStateDTO) dto.getApplianceState()).getConnectionState())
+                        ? OnOffType.from(true)
+                        : OnOffType.from(false);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    public Map<String, String> refreshProperties() {
+        Map<String, String> properties = new HashMap<>();
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ElectroluxAppliancesBridgeHandler bridgeHandler = (ElectroluxAppliancesBridgeHandler) bridge.getHandler();
+            if (bridgeHandler != null) {
+                ApplianceDTO dto = bridgeHandler.getElectroluxAppliancesThings().get(config.getSerialNumber());
+                if (dto != null) {
+                    properties.put(Thing.PROPERTY_VENDOR, dto.getApplianceInfo().getApplianceInfo().getBrand());
```

Comment:
Better to short cache the data, also easier to read. Applies to rest of the poperties.put lines..
```suggestion
                    var applianceInfo = dto.getApplianceInfo().getApplianceInfo();
                    properties.put(Thing.PROPERTY_VENDOR, applianceInfo.getBrand());
```

Reply:
Nice use of var! :-)